### DISCUSSION TITLE
is it worth adding a dependency for gcp?

### DISCUSSION THREAD

**Discussion by glats** [micronaut-projects/micronaut-core#6500] [Ideas] [answered]
I created a project with the following features:
 - graalvm
 - reactor
 - lombok
 - gcp-pubsub
 - testcontainers

When creating a native image I got the following error:
```shell
To see how the classes got initialized, use --trace-class-initialization=org.conscrypt.Conscrypt,org.conscrypt.OpenSSLProvider
analysis: 107,698.48 ms,  5.59 GB
Error: Classes that should be initialized at run time got initialized during image building:
 org.conscrypt.Conscrypt was unintentionally initialized at build time. To see why org.conscrypt.Conscrypt got initialized use --trace-class-initialization=org.conscrypt.Conscrypt
org.conscrypt.OpenSSLProvider was unintentionally initialized at build time. To see why org.conscrypt.OpenSSLProvider got initialized use --trace-class-initialization=org.conscrypt.OpenSSLProvider
```

Then i made a little research and I found this [answer](https://stackoverflow.com/a/65505573). So adding this dependency 
```xml
    <dependency>
      <groupId>com.google.cloud</groupId>
      <artifactId>native-image-support</artifactId>
      <version>0.9.0</version>
    </dependency>
```
the native imagen was made well.

my question: is it worth adding this dependency to the micronaut project generator?

Thanks.

**Comment by pgressa** [comment, marked as answer]
Hi I would recommend to create an issue in https://github.com/micronaut-projects/micronaut-gcp first. Micronaut should be native image compatible by default so we need to find out whether this shouldn't be a default dependency.

**Reply by graemerocher** [reply]
Yeah seems like a good improvement to add

**Reply by glats** [reply]
done! thanks!

**Comment by glats** [comment]
Finally added https://github.com/micronaut-projects/micronaut-gcp/pull/682 !
